File: 3095/main.go
package main

import (
	"fmt"
)

func main() {
	fmt.Println(minimumSubarrayLength([]int{1,2,3}, 2))		// 1
	fmt.Println(minimumSubarrayLength([]int{8,2,1}, 10))	// 3
	fmt.Println(minimumSubarrayLength([]int{32,2,24,1}, 35))	// 3
}

func minimumSubarrayLength(nums []int, k int) int {
	n := len(nums)
	res := n+1
	for i := 0; i < n; i++ {
		v := 0
		for j := i; j < n; j++ {
			v |= nums[j]
			if v >= k {
				res = min(res, j-i+1)
				break
			}
		}
	}
	if res == n + 1{
		return -1
	}
	return res
}

File: 3095/main.rs
struct Solution{}

impl Solution {
    pub fn minimum_subarray_length(nums: Vec<i32>, k: i32) -> i32 {
        let n = nums.len();
        let mut res = n + 1;
        for i in 0..n {
            let mut v = 0;
            for j in i..n {
                v |= nums[j];
                if v >= k {
                    res = std::cmp::min(res, j-i+1);
                    break
                }
            }
        }
        if res == n+1 {
            return -1
        }
        res as i32
    }
}

fn main() {
    println!("{:#?}", Solution::minimum_subarray_length(vec![1,2,3], 2));
}